spec/formatting.md

```diff
@@ -618,13 +640,18 @@ appropriate data type or structure. Some examples of these include:
 - A string with associated attributes for portions of its text.
 - A flat sequence of objects corresponding to each resolved value.
 - A hierarchical structure of objects that group spans of resolved values,
-  such as sequences delimited by "open" and "close" _function_ _annotations_.
+  such as sequences delimited by _markup-open_ and _markup-close_ _placeholders_.
 
 Implementations SHOULD provide _formatting_ result types that match user needs,
 including situations that require further processing of formatted messages.
 Implementations SHOULD encourage users to consider a formatted localised string
 as an opaque data structure, suitable only for presentation.
 
+When formatting to a string, the default representation of all _markup_
+MUST be an empty string.
+Implementations MAY offer functionality for customizing this,
+such as by emitting XML-ish tags for each _markup_.
```

Comment:
I would tighten this up a bit. The normative text is appropriate, but I would avoid the MUST. I would also make the example a bit more fleshed out.

```suggestion
The default representation of all types of _markup_ placeholder
is an empty string.
That is, unless otherwise customized, open, close and standalone
markup placeholders are omitted from the output.
Implementations MAY offer functionality to customize this behavior.

> For example, an implementation might replace markup with
> styling attributes in an attributed string API 
> or it might replace the placeholders with HTML markup.
```

Reply:
Your suggestion leaves out the "When formatting to a string" qualifier. Is that intentional?

Reply:
Yes.